PIL/Image.py

```diff
@@ -1587,32 +1591,59 @@ def rotate(self, angle, resample=NEAREST, expand=0):
         if angle == 270 and expand:
```

Comment:
The fastpath cases aren't properly handled except for the angle == 0 case, which is correctly still a no-op. If the caller has passed a center or translate for the other three cases, the result is going to be different than the fast path.